airbyte-integrations/connectors/source-github/source_github/streams.py

```diff
@@ -0,0 +1,405 @@
+            yield {"pull_request_number": pull_request["number"]}
+
+
+class Collaborators(GithubStream):
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return f"repos/{self.repository}/collaborators"
+
+
+class Releases(GithubStream):
+    object_fields = ("author",)
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return f"repos/{self.repository}/releases"
+
+    def transform(self, record: Mapping[str, Any]) -> Mapping[str, Any]:
+        record = super().transform(record=record)
+
+        assets = record.get("assets", [])
+        for asset in assets:
+            uploader = asset.pop("uploader", None)
+            asset["uploader_id"] = uploader.get("id") if uploader else None
+
+        return record
+
+
+class Events(GithubStream):
+    object_fields = (
+        "actor",
+        "repo",
+        "org",
+    )
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return f"repos/{self.repository}/events"
+
+
+class Comments(GithubStream):
+    object_fields = ("user",)
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return f"repos/{self.repository}/issues/comments"
+
+
+class PullRequests(GithubStream):
+    object_fields = (
+        "user",
+        "milestone",
+        "assignee",
+    )
+    list_fields = (
+        "labels",
+        "assignees",
+        "requested_reviewers",
+        "requested_teams",
+    )
+
+    def read_records(self, **kwargs) -> Iterable[Mapping[str, Any]]:
+        with vcr.use_cassette(cache_file.name, record_mode="new_episodes", serializer="json"):
+            yield from super().read_records(**kwargs)
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return f"repos/{self.repository}/pulls"
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        params = super().request_params(stream_state=stream_state, stream_slice=stream_slice, next_page_token=next_page_token)
+        params["state"] = "all"
+        return params
+
+    def transform(self, record: Mapping[str, Any]) -> Mapping[str, Any]:
+        record = super().transform(record=record)
+
+        head = record.get("head", {})
+        head_user = head.pop("user", None)
+        head["user_id"] = head_user.get("id") if head_user else None
```

Comment:
```suggestion
        head_user = head.pop("user", {})
        head["user_id"] = head_user.get("id")
```

Reply:
But my proposal is failing, if there could be an option in which:
```
{
    "user": None,
    ....
}
```

Reply:
This won't work in cases when `user` field has `null` value. When `user` has `null` value suggested code will throw `AttributeError: 'NoneType' object has no attribute 'get'`.